site/content/contribute/desktop/code-signing.md

```diff
@@ -0,0 +1,188 @@
+---
+title: "Code Signing"
+date: 2018-01-16T10:32:51-05:00
+subsection: "Desktop App"
+---
+# Code Signing
+When releasing the Mattermost Desktop application for Windows and macOS, we have to sign the executable with a certificate that allows the end users' computer to verify our identity. 
+
+The signing procedure varies depending on the platform that the release is destined for, and the platform that the signing operation is performed on. This page attempts to document the procedure and some common pitfalls that developers may encounter along the way.
+
+## Code Signing Windows Releases
+Releases destined for Windows can be code signed on Windows, macOS, or Linux. 
+
+#### Prerequisites
+In order to code sign releases on behalf of Mattermost Inc., you'll need a `.pfx` (Personal Information Exchange) file that contains Mattermost's public key (SSL certificate file), and the associated private key file. The file is protected by a password that you'll need in order to use it. 
+
+This file has been shared in LastPass. Talk to Joram Wilander, Corey Hulen, or Jonathan Fritz to get access to it. 
+
+#### Code Signing on Windows
+On Windows hosts, Microsoft's [`signtool`](https://msdn.microsoft.com/en-us/library/windows/desktop/aa387764(v=vs.85).aspx) utility can be used to code sign releases.
+
+#### Code Signing on macOS and Linux
+On macOS and Linux hosts, the open source [`osssigntool`](https://sourceforge.net/projects/osslsigncode/) can be used to code sign releases. It can be installed via Homebrew on macOS:
+```bash
+~$ brew install openssl
+~$ brew install osslsigncode
+```
+or via Apt on Ubuntu:
+```bash
+~$ sudo apt-get install openssl
+~$ sudo apt-get install osslsigncode
+```
+
+After successfully [building and packaging](/contribute/desktop/developer-setup#building) the Mattermost Desktop application for Windows, you can run the signing command from the root of the repository:
+```bash
+~$ osslsigncode -pkcs12 [PATH_TO_THE_PFX_FILE] -pass [PFX_FILE_PASSWORD] -n "Mattermost Desktop" -i https://mattermost.com -t http://timestamp.verisign.com/scripts/timstamp.dll -h sha2 -in [PATH_TO_UNSIGNED_EXE] -out [PATH_TO_WRITE_SIGNED_EXE_TO]
+Succeeded
+```
+Where
+<ul>
+    <li>**PATH_TO_THE_PFX_FILE** is the absolute path to the `.pfx` file that was obtained in the Prerequisites section above</li>
+    <li>**PFX_FILE_PASSWORD** is the password that protects the `.pfx` file from being misused</li>
+    <li>**PATH_TO_UNSIGNED_EXE** is the absolute path of the unsided executable that you want to sign. It is typically in the `release/win` or `release/win-ia32` subdirectory of the repository</li>
```

Comment:
> unsided executable

I assume that should be "unsigned executable"?

Reply:
Yup